### PR TITLE
Make E2E tests more robust

### LINKED ISSUE #229: Regenerate "Golden" End-to-end Test Logs
**Description:**

The end-to-end tests rely on previously generated "golden" log files to compare current log output from a pipeline run on the cluster with the log output of a previous "golden" pipeline run.

Some of the "golden" log files appear to no longer be current since several code changes have modified the log output of the pipeline runs.

**What needs to be done:**

Set up a Kubernetes cluster with Kubeflow and Tekton and regenerate the "golden" log files using the following command:

```Bash
make e2e_test GENERATE_GOLDEN_E2E_LOGS=True
```

**Additional Information:**

For the test using `ResourceOp`s to success it is required create a `ClusterRoleBinding` with `cluster-admin` to the `default` service account. Since the E2E tests are run in the `kubeflow` namespace, the `default` service account is deployed it to the `kubeflow` namespace.

Create RBAC permission by running the following command:

```Bash
cat <<EOF |kubectl apply -f -
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  name: default-admin
subjects:
  - kind: ServiceAccount
    name: default
    namespace: kubeflow
roleRef:
  kind: ClusterRole
  name: cluster-admin
  apiGroup: rbac.authorization.k8s.io
EOF
```

After running that command you should see this message:

```
clusterrolebinding.rbac.authorization.k8s.io/default-admin created
```

**Environment:**

* Python Version (use `python --version`): 3.5.3 and up
* SDK Version: `master`
* Tekton Version (use `tkn version`): `0.14.0` or later
* Kubernetes Version (use `kubectl version`): `0.16.0`

### PR DESCRIPTION
**Pull request by ckadner**
**Description of your changes:**

* Tolerate erratic log lines in E2E tests
* Add option to re-run only tests that failed previously
* Add option to exclude/include certain test_cases only
* Add `volume_op.yaml` to ignored tests as the created PVC
  needs to be deleted before/after test run
* Regenerate "golden" log files after recent commits
  changed the log output regarding artifacts

**Environment tested:**

* Python Version (use `python --version`): `3.7.5`
* Tekton Version (use `tkn version`): `0.14.0`
* Kubernetes Version (use `kubectl version`): `v1.16.11+IKS`
* OS (e.g. from `/etc/os-release`): Mac OS 10.15

**Additional Information:**

Closes #229

/cc @Tomcli 

@moficodes

### CODE REVIEW

**Comment by Tomcli** [comment]
Thanks @ckadner, the log outcomes can be different based on the tkn version we are using, can we add an extra function in the script to make sure they use the correct tkn version for the e2e tests? Thanks

**Comment by Tomcli** [comment]
and we are using tkn 0.10.0 for the tests?

**Comment by ckadner** [comment]
> and we are using `tkn` `0.10.0` for the tests?

```YAML
Client version: 0.10.0
Pipeline version: v0.14.0
Triggers version: unknown
```

**Comment by ckadner** [comment]
> Thanks @ckadner, the log outcomes can be different based on the tkn cli version we are using, can we add an extra function in the script to make sure they use the correct tkn cli version for the e2e tests? Thanks

I added the required/expected versions and checks in `Makefile` and in `compiler_tests_e2e.py`. They can be overridden with environment variables, if need be until changed in a subsequent version upgrade.

```Bash
make e2e_test TKN_PIPELINE_VERSION=0.14 TKN_CLIENT_VERSION=0.10
```

File: Makefile [hunk 1]
```diff
@@ -19,6 +19,9 @@ VENV ?= .venv
 export VIRTUAL_ENV := $(abspath ${VENV})
 export PATH := ${VIRTUAL_ENV}/bin:${PATH}
 
+TKN_PIPELINE_VERSION ?= "0.14"
```
Comment (Tomcli) [inline Makefile:22]:
```suggestion
TKN_PIPELINE_VERSION ?= "0.14.0"
```

File: Makefile [hunk 2]
```diff
@@ -19,6 +19,9 @@ VENV ?= .venv
...
+TKN_CLIENT_VERSION ?= "0.10"
```
Comment (Tomcli) [inline Makefile:23]:
```suggestion
TKN_CLIENT_VERSION ?= "0.10.0"
```

**Comment by ckadner** [comment]
@moficodes -- if you get a chance, could you checkout this PR? Thanks :-)

**Comment by ckadner** [comment]
@Tomcli -- I left the service version parts off because I expect no change in the behavior of the `tkn` command output for service version fix releases. This way the E2E tests won't break each time there is a service version patch and `v0.14.1`, `v0.14.2` ... don't require a change to our E2E tests.

**Comment by Tomcli** [comment]
@moficodes any comment you want to add on top of this?

**Pull request merged**